🐛[BUG]: Pytorch import error in register_prop_rule

### Version

main

### On which installation method(s) does this occur?

_No response_

### Describe the issue

In the domain parallel we have

https://github.com/NVIDIA/physicsnemo/blob/main/physicsnemo/domain_parallel/custom_ops/_tensor_ops.py#L42

```
if check_version_spec("torch", "2.10.0"):
    from torch.distributed.tensor._ops.registration import (
        register_prop_rule,
    )
else:
    from torch.distributed.tensor._ops.utils import (
        register_prop_rule,
    )
```

This causes issues with anything no exactly 2.10.0
Like the PyTorch containers from nvidia.

We need to have a check thats probably anything greater than 2.9.1

### Minimum reproducible example

```shell
nvcr.io/nvidia/pytorch:26.01-py3
```

which has torch 2.10.0a0+a36e1d39eb.nv26.01.42222806

### Relevant log output

```shell
ImportError while importing test module '/app/test/models/dit/test_dit.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/models/dit/test_dit.py:25: in <module>
    from physicsnemo.experimental.models.dit import DiT
physicsnemo/experimental/models/dit/__init__.py:17: in <module>
    from .layers import (
physicsnemo/experimental/models/dit/layers.py:62: in <module>
    from physicsnemo.domain_parallel import ShardTensor
physicsnemo/domain_parallel/__init__.py:66: in <module>
    register_custom_ops()
physicsnemo/domain_parallel/__init__.py:55: in register_custom_ops
    from .custom_ops import (
physicsnemo/domain_parallel/custom_ops/__init__.py:24: in <module>
    from ._tensor_ops import unbind_rules
physicsnemo/domain_parallel/custom_ops/_tensor_ops.py:47: in <module>
    from torch.distributed.tensor._ops.utils import (
E   ImportError: cannot import name 'register_prop_rule' from 'torch.distributed.tensor._ops.utils' (/usr/local/lib/python3.12/dist-packages/torch/distributed/tensor/_ops/utils.py)
```

### Environment details

```shell

```

--- a/physicsnemo/domain_parallel/custom_ops/_tensor_ops.py
+++ b/physicsnemo/domain_parallel/custom_ops/_tensor_ops.py
@@ -39,7 +39,7 @@ from physicsnemo.domain_parallel._shard_tensor_spec import (
     _stride_from_contiguous_shape_C_style,
 )
 
-if check_version_spec("torch", "2.10.0"):
+if check_version_spec("torch", "2.10.0a"):
     from torch.distributed.tensor._ops.registration import (
         register_prop_rule,
     )

